torch/_higher_order_ops/while_loop.py

```diff
@@ -246,16 +321,41 @@ def _trace_while_loop(
 
 
 def check_outputs_carry_consistency(
-    outs: List[torch.Tensor], carries: List[torch.Tensor]
+    outs: List[Union[torch.Tensor, torch.SymInt, int]],
+    carries: List[Union[torch.Tensor, torch.SymInt, int]],
 ) -> None:
-    all_diffs_in_meta = []
-    for out, cry in zip(outs, carries):
-        if diff := diff_tensor_meta(
-            _extract_tensor_metadata(cry), _extract_tensor_metadata(out)
-        ):
-            all_diffs_in_meta.append(",".join(diff))
-    if all_diffs_in_meta:
-        diff_str = "\n".join(all_diffs_in_meta)
+    def diff_meta_pairs(
+        lhs_list: List[Union[torch.Tensor, torch.SymInt, int]],
+        rhs_list: List[Union[torch.Tensor, torch.SymInt, int]],
+    ) -> List[str]:
+        def diff_meta(
+            lhs: Union[torch.Tensor, torch.SymInt, int],
+            rhs: Union[torch.Tensor, torch.SymInt, int],
```

Comment:
This is kind of difficult to read. What happens when lhs is a Tensor and rhs is an int? What happens if lhs is a SymInt and rhs is an int?

Reply:
I would expect that Tensor - Tensor is fine, and then (symint, int) - (symint - tensor) is fine. Tensor - (symint, int) should probably error

Reply:
yeah, later of the stack tightens the check. I can move it here.